### PR TITLE
fix(wsgi): Log SystemExit (with non-zero exit code) and KeyboardInterrupt

### PR DESCRIPTION
**Pull request by saifelse**
Neither SystemExit nor KeyboardInterrupt subclass Exception, so we must
explicitly handle these two exception classes in addition to Exception in the
WSGI middleware.

This is notably a direct port from raven-python:
 - https://github.com/getsentry/raven-python/blob/master/raven/middleware.py
 - https://github.com/getsentry/raven-python/blob/master/tests/middleware/tests.py

Fixes: GH-379

There are notably 3 places an exception could be raised, but it looks like we're only
testing one case. If we want to test the other two, would appreciate advice 🙏🏾.

### CODE REVIEW

**Comment by saifelse** [comment]
@untitaker : thanks, back to you!

File: sentry_sdk/integrations/wsgi.py [hunk 1]
```diff
@@ -192,8 +197,8 @@ def close(self):
                 self._response.close()
             except AttributeError:
                 pass
-            except Exception:
-                reraise(*_capture_exception(self._hub))
+            except BaseException:
```
Comment (untitaker) [inline sentry_sdk/integrations/wsgi.py]: You're missing an `as e` here

Comment (saifelse) [reply sentry_sdk/integrations/wsgi.py]: oof, thanks

**Comment by saifelse** [comment]
> one change necessary, also please make sure that CI passes (see `CONTRIBUTING.md`)
looks like it was failing from zeus, so hopefully merging in master takes care of that 🤞🏾

File: sentry_sdk/integrations/wsgi.py [hunk 2]
```diff
@@ -149,17 +149,22 @@ def get_client_ip(environ):
     return environ.get("REMOTE_ADDR")
 
 
-def _capture_exception(hub):
-    # type: (Hub) -> ExcInfo
+def _capture_exception(hub, e):
```
Comment (untitaker) [inline sentry_sdk/integrations/wsgi.py]: Instead of taking an additional param here, could you read this info from `exc_info[2]`?

**Comment by untitaker** [comment]
Thanks!

**Pull request merged**